### PR TITLE
Add headings to conditional table when collapsed

### LINKED ISSUE #4627: Conditional table format missing column headings
After choosing to make a format conditional based on a field, you immediately see the collapsed grid view. The first field is supposed to contain the conditional value the user needs to enter for that field (i.e. "Role"). There are no headings and no indication of what that is or how to configure this.

http://localhost/specify/resources/app-resource/9/formatter/AccessionAgent/1

image

When expanded, more information is given:
image

**Expected Behavior:**
There should be column headings to indicate what is happening
image

### PR DESCRIPTION
**Pull request by CarolineDenis**
Fixes #4627

### Checklist

- [x] Self-review the PR after opening it to make sure the changes look good
      and self-explanatory (or properly documented)
- [x] Add relevant issue to release milestone

### Testing instructions

- go in app resources 
- choose records formatters 
- choose a table that has formatters

- [ ] verify that there is 3 headers when the conditional is collapsed 
- [ ] verify that the headers are removed when expanding

### CODE REVIEW

**Review by pashiav — APPROVED** [review]
### Testing instructions

- go in app resources 
- choose records formatters 
- choose a table that has formatters

- [X] verify that there is 3 headers when the conditional is collapsed 
- [X] verify that the headers are removed when expanding 


https://github.com/specify/specify7/assets/118556884/f490e366-3ee5-4950-bc2e-e33e9dde566d

File: specifyweb/frontend/js_src/lib/components/Formatters/Definitions.tsx [hunk 1]
```diff
@@ -218,23 +229,30 @@ function ConditionalFormatter({
...
-          <Button.Icon
-            icon="chevronDown"
-            title={resourcesText.expandConditionalField()}
-            onClick={handleToggle}
-          />
+      <div className="flex flex-col p-2">
+        {!isExpanded && index === 0 ? (
+          <span className="font-bold">{commonText.expand()}</span>
         ) : null}
+        <div className="flex justify-end">
```
Comment (melton-jason) [inline specifyweb/frontend/js_src/lib/components/Formatters/Definitions.tsx:236]: (optional)
I don't think `justify-end` is needed here. 
The elements being "right justified" should be handled by the above span with `-ml-2`, correct? 
Unless there is some other reason `justify-end` is used here?

**Review by melton-jason — APPROVED** [review]
A lot of the changes here are purely relating to styling, so I will leave the design to @specify/ux-testing to approve/deny. 

Although I did additionally quickly encounter the following. Is this known? It was not caused by these changes, or at the very least also happened before them.

https://github.com/specify/specify7/assets/64045831/358eb1ea-babb-449b-8d14-0db39d1a7782

**Comment by CarolineDenis** [comment]
> A lot of the changes here are purely relating to styling, so I will leave the design to @specify/ux-testing to approve/deny.
> 
> Although I did additionally quickly encounter the following. Is this known? It was not caused by these changes, or at the very least also happened before them.
> 
>  Screen.Recording.2024-03-11.at.2.14.11.PM.mov

yes it is known and in an other issue. But still, Thanks for catching it!

File: specifyweb/frontend/js_src/lib/components/Formatters/Definitions.tsx (hunk 1 above)
Comment (CarolineDenis) [reply specifyweb/frontend/js_src/lib/components/Formatters/Definitions.tsx:236]: the justify end is needed here to keep the arrow at the same place when there is no trash icon
see with and without

https://github.com/specify/specify7/assets/108160931/22ec8bc0-f6a0-41e9-8d3f-628b4a990013


https://github.com/specify/specify7/assets/108160931/780be95b-da6e-4dda-82d5-ba6fbb9a5636

**Review by emenslin — APPROVED** [review]
### Testing instructions

- go in app resources 
- choose records formatters 
- choose a table that has formatters

- [x] verify that there is 3 headers when the conditional is collapsed 
- [x] verify that the headers are removed when expanding 

Looks good!
Screenshot 2024-03-12 090712

**Pull request merged**